### ISSUE TITLE
v1.0.13 introduced an error if `trivial_numeric_casts` is denied

### ISSUE THREAD

**Issue by matthiasbeyer** [serde-rs/serde#1051]
Hi,

I updated `cargo update`d on my machine and got the new serde versions:

```
    Updating serde v1.0.12 -> v1.0.13
    Updating serde_derive v1.0.12 -> v1.0.13
    Updating serde_derive_internals v0.15.1 -> v0.16.0
```

and now my code fails to compile because I turned on `trivial_numeric_casts` denying:

```

error: trivial numeric cast: `u64` as `u64`. Cast can be replaced by coercion, this might require type ascription or a temporary variable
  --> src/file_abstraction/stdio/mapper/json.rs:35:17
   |
35 | #[derive(Debug, Deserialize, Serialize)]
   |                 ^^^^^^^^^^^
   |
note: lint level defined here
  --> src/lib.rs:27:5
   |
27 |     trivial_numeric_casts,
   |     ^^^^^^^^^^^^^^^^^^^^^

error: trivial numeric cast: `u64` as `u64`. Cast can be replaced by coercion, this might require type ascription or a temporary variable
  --> src/file_abstraction/stdio/mapper/json.rs:55:17
   |
55 | #[derive(Debug, Deserialize, Serialize)]
   |                 ^^^^^^^^^^^

error: aborting due to 2 previous errors

error: Could not compile `libimagstore`.

To learn more, run the command again with --verbose.
```

The repository of my code [is here](https://github.com/matthiasbeyer/imag).

**Comment by dtolnay** [comment]
Thanks for reporting this. I published a fix in 1.0.14.

**Comment by matthiasbeyer** [comment]
HOLY! I did not expect this to be fixed _so fast_! Thank you very much, you just made my day!
